File: ecos-records/src/main/java/ru/citeck/ecos/records2/predicate/model/EmptyPredicate.java
package ru.citeck.ecos.records2.predicate.model;

import ecos.com.fasterxml.jackson210.annotation.JsonProperty;

import java.util.Collections;
import java.util.List;
import java.util.Objects;

public class EmptyPredicate extends AttributePredicate {

    private static final String TYPE = "empty";

    public EmptyPredicate() {
    }

    public EmptyPredicate(String attribute) {
        setAttribute(attribute);
    }

    @JsonProperty("t")
    public String getType() {
        return TYPE;
    }

    public static List<String> getTypes() {
        return Collections.singletonList(TYPE);
    }

    @Override
    public <T extends Predicate> T copy() {

        @SuppressWarnings("unchecked")
        T result = (T) new EmptyPredicate(getAttribute());

        return result;
    }

    @Override
    public String toString() {
        return "EMPTY:'" + getAttribute() + "'";
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        EmptyPredicate that = (EmptyPredicate) o;
        return Objects.equals(getAttribute(), that.getAttribute());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getAttribute());
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records2/predicate/model/EmptyPredicate.kt
package ru.citeck.ecos.records2.predicate.model

import ecos.com.fasterxml.jackson210.annotation.JsonProperty

class EmptyPredicate : AttributePredicate {

    companion object {

        const val TYPE = "empty"

        @JvmStatic
        fun getTypes(): List<String> {
            return listOf(TYPE)
        }
    }

    constructor()

    constructor(attribute: String?) {
        setAttribute(attribute)
    }

    @JsonProperty("t")
    fun getType(): String {
        return TYPE
    }

    override fun <T : Predicate> copy(): T {
        @Suppress("UNCHECKED_CAST")
        return EmptyPredicate(getAttribute()) as T
    }

    override fun toString(): String {
        return "{\"t\":\"$TYPE\",\"att\":\"${getAttribute()}\"}"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        other as EmptyPredicate
        return getAttribute() == other.getAttribute()
    }

    override fun hashCode(): Int {
        return getAttribute().hashCode()
    }
}
